modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java

```diff
@@ -518,11 +567,20 @@ public ObjectEntry expireObjectEntryByVersion(
 			String scopeKey, int version)
 		throws Exception {
 
-		return _expireObjectEntryVersion(
-			dtoConverterContext, objectDefinition,
+		com.liferay.object.model.ObjectEntry serviceBuilderObjectEntry =
 			_objectEntryService.getObjectEntry(
 				externalReferenceCode, getGroupId(objectDefinition, scopeKey),
-				objectDefinition.getObjectDefinitionId()),
+				objectDefinition.getObjectDefinitionId());
+
+		if (serviceBuilderObjectEntry.getStatus() ==
+				WorkflowConstants.STATUS_IN_TRASH) {
+
+			throw new ObjectEntryVersionStatusException.
+				MustNotExpireObjectEntryVersionInTrash();
+		}
+
+		return _expireObjectEntryVersion(
+			dtoConverterContext, objectDefinition, serviceBuilderObjectEntry,
 			version);
```

Comment:
@yuriymr please move this logic to `ObjectEntryVersionLocalServiceImpl` -> `expireObjectEntryVersion`